include/secp256k1_musig.h

```diff
@@ -358,14 +357,72 @@ SECP256K1_API int secp256k1_musig_nonce_gen(
     const secp256k1_context *ctx,
     secp256k1_musig_secnonce *secnonce,
     secp256k1_musig_pubnonce *pubnonce,
-    const unsigned char *session_id32,
+    const unsigned char *session_secrand32,
     const unsigned char *seckey,
     const secp256k1_pubkey *pubkey,
     const unsigned char *msg32,
     const secp256k1_musig_keyagg_cache *keyagg_cache,
     const unsigned char *extra_input32
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(6);
 
+
+/** Alternative way to generate a nonce and start a signing session
+ *
+ *  This function outputs a secret nonce that will be required for signing and a
+ *  corresponding public nonce that is intended to be sent to other signers.
+ *
+ *  This function differs from `secp256k1_musig_nonce_gen` by accepting a
+ *  non-repeating counter value instead of a secret random value. This requires
+ *  the seckey argument to be mandatory.
+ *
+ *  MuSig differs from regular Schnorr signing in that implementers _must_ take
+ *  special care to not reuse a nonce. This can be ensured by following these rules:
+ *
+ *  1. The nonrepeating_cnt argument must be a counter value that never
+ *     repeats, i.e., you must never call `secp256k1_musig_nonce_gen_counter`
+ *     twice with the same seckey and nonrepeating_cnt value.
+ *  2. If you already know the seckey, message or aggregate public key
+ *     cache, they can be optionally provided to derive the nonce and increase
+ *     misuse-resistance. The extra_input32 argument can be used to provide
```

Comment:
seckey is mandatory for this function anyway.

I think usually I would suggest moving this big explainer at the top of the module, so that it covers both functions and does not need to be repeated. But in this case of a big fat warning, it's probably better to keep it in the docstring, where it's harder to miss...

Reply:
dropped mention of seckey